plugin/sublime-text/plugin_subl3.py

```diff
@@ -0,0 +1,152 @@
+        return self.buffers[id_]
+
+
+class QuickLintJsListener(sublime_plugin.ViewEventListener):
+    """Listens for events bound to a specific view."""
+
+    # The internal strategy used is to share information between
+    # all views that belong to the same buffer. Because that way,
+    # if there are multiple views (tabs) of the same buffer (file),
+    # they will all apply the same changes (have squiggly underlines
+    # and pop-ups available):
+    #
+    # https://github.com/quick-lint/quick-lint-js/pull/328#issuecomment-869038036
+
+    buffers_manager = BuffersManager()
+
+    @classmethod
+    def is_applicable(cls, settings):
+        syntax = settings.get("syntax", "")
+        if "JavaScript" in syntax:
+            return True
+        return False
+
+    @classmethod
+    def applies_to_primary_view_only(cls):
+        return False
+
+    def __init__(self, view):
+        super().__init__(view)
+        self.buffers_manager.add_view(self.view)
+        self.buffer = self.buffers_manager.get_buffer(self.view)
+        self.on_modified()
+
+    def __del__(self):
+        self.buffers_manager.remove_view(self.view)
+
+    def on_load(self):
+        self.on_modified()
+
+    def on_modified(self):
+        self.buffer.parser.set_text()
+        self.buffer.parser.lint()
+        self.add_squiggly_underlines()
+
+    def on_hover(self, point, hover_zone):
+        if hover_zone == sublime.HOVER_TEXT:
+            for diagnostic in self.buffer.parser.diagnostics:
+                # If the user hovers over the diagnostic region
+                # (region with squiggly underlines).
+                if diagnostic.region.contains(point):
+                    self.add_popup(diagnostic)
+
+    def add_squiggly_underlines(self):
+        warning_regions, error_regions = self.get_severity_regions()
+        flags = (
+            sublime.DRAW_SQUIGGLY_UNDERLINE
+            | sublime.DRAW_NO_FILL
+            | sublime.DRAW_NO_OUTLINE
+        )
+        for view in self.buffer.views:
+            view.add_regions("2", warning_regions, "region.orangish", "", flags)
+            view.add_regions("1", error_regions, "region.redish", "", flags)
+
+    def get_severity_regions(self):
+        warning_regions = []
+        error_regions = []
+        for diagnostic in self.buffer.parser.diagnostics:
+            if c_api.SeverityEnumeration.WARNING == diagnostic.severity:
+                warning_regions.append(diagnostic.region)
+            elif c_api.SeverityEnumeration.ERROR == diagnostic.severity:
+                error_regions.append(diagnostic.region)
+        return warning_regions, error_regions
+
+    def add_popup(self, diagnostic):
+        minihtml = """
+        <span>%s
+            <span style=\"color: %s;\">quick-lint-js(%s)</span>
+        </span>
+        """
+        color = self.view.style_for_scope("comment.line")["foreground"]
+        content = minihtml % (diagnostic.message, color, diagnostic.code)
```

Comment:
What if quick-lint-js' diagnostic contains `<`? We should HTML-escape the message and the code (and probably the color).